### PR TITLE
Feature: Implement Wizard pages

### PR DESCRIPTION
**Pull request by akhlopiachyi**

### CODE REVIEW

File: src/modules/admin/markets/actions.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+});
+
+export const getMarketsListData = (payload: MarketsListData['payload']): MarketsListData => ({
+    type: MARKETS_LIST_DATA,
+    payload,
+});
+
+export const getMarketsListError = (): MarketsListError => ({
+    type: MARKETS_LIST_ERROR,
+});
```
Comment (oyershov) [inline src/modules/admin/markets/actions.ts:106]: Why actions mixed? Please group correctly.

File: src/api/types.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ declare global {
         captcha_id?: string;
         password_min_entropy: number;
         palette?: string;
+        wizard_step?: number;
```
[the remark this answers is not in the document]
Comment (oyershov) [reply src/api/types.ts]: fixed

**Comment by oyershov** [comment]
please, fix tests

**Review by oyershov — APPROVED** [review]

File: src/assets/images/setup/ArrowAllRightIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+
+interface ArrowAllRightIconProps {
+    className?: string;
+}
+
+export const ArrowAllRightIcon: React.FC<ArrowAllRightIconProps> = (props: ArrowAllRightIconProps) => (
```
Comment (josadcha) [inline src/assets/images/setup/ArrowAllRightIcon.tsx:7]: you've defined className prop but haven't use it

File: src/assets/images/setup/ArrowLeftIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+interface ArrowLeftIconProps {
+    className?: string;
+}
+
+export const ArrowLeftIcon: React.FC<ArrowLeftIconProps> = (props: ArrowLeftIconProps) => (
```
Comment (josadcha) [inline src/assets/images/setup/ArrowLeftIcon.tsx:7]: you've defined className prop but haven't use it

File: src/assets/images/setup/ArrowRightIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+interface ArrowRightIconProps {
+    className?: string;
```
Comment (josadcha) [inline src/assets/images/setup/ArrowRightIcon.tsx:4]: same

File: src/assets/images/setup/CloseSetupIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+interface CloseSetupIconProps {
+    className?: string;
```
Comment (josadcha) [inline src/assets/images/setup/CloseSetupIcon.tsx:4]: same

File: src/components/SetupComponents/SetupCongratsBlock/index.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+export class SetupCongratsBlock extends React.Component {
+    public render() {
+        return (
+            <div className="setup-congrats-block">
+                <img src={congratsImage} alt="platform-setup-congrats" />
+            </div>
+        );
+    }
+}
```
Comment (josadcha) [inline src/components/SetupComponents/SetupCongratsBlock/index.tsx:12]: better to use stateless component here

File: src/components/SetupComponents/SetupFormBlock/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+
+export interface SetupFormBlockProps {
+    title: string;
+    subtitle: string;
+    children?: React.ReactNode;
+}
+
+export class SetupFormBlock extends React.Component<SetupFormBlockProps> {
```
Comment (josadcha) [inline src/components/SetupComponents/SetupFormBlock/index.tsx:9]: same here, better to use functional component according to our plan about moving Baseapp to func

File: src/components/SetupComponents/SetupGeneralSettingsForm/index.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    public render() {
+        const {  exchangeName, exchangeUrl } = this.state;
+
+        return (
+            <React.Fragment>
+                <form className="setup-general-settings-form">
+                    <SetupFormInput
+                        label="Exchange Name"
+                        value={exchangeName}
+                        tooltipText="Consequat cupidatat officia duis duis enim voluptate cillum consectetur amet qui commodo proident elit."
```
Comment (josadcha) [inline src/components/SetupComponents/SetupGeneralSettingsForm/index.tsx:33]: add translations please

File: src/components/SetupComponents/SetupGeneralSettingsForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                        handleChangeInput={this.handleChangeExchangeName}
+                    />
+                    <SetupFormInput
+                        label="Exchange Url"
+                        value={exchangeUrl}
+                        tooltipText="Consequat cupidatat officia duis duis enim voluptate cillum consectetur amet qui commodo proident elit."
```
Comment (josadcha) [inline src/components/SetupComponents/SetupGeneralSettingsForm/index.tsx:39]: same

File: src/components/SetupComponents/SetupMarketsBlock/index.tsx [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+            selectedAvailableMarkets: [],
+            selectedAddedMarkets: [],
+        };
+    }
+
+    public componentDidMount() {
+        this.props.fetchMarkets();
+    }
+
+    public componentWillReceiveProps(nextProps: SetupMarketBlockProps) {
```
Comment (josadcha) [inline src/components/SetupComponents/SetupMarketsBlock/index.tsx:38]: if you don't have enough time to rewrite to functional, at least use componentDidUpdate please

File: src/components/SetupComponents/SetupMarketsBlock/index.tsx [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+                    <div className="setup-markets-block__list">
+                        <div className="setup-markets-block__list__title">Selected to your platform</div>
+                        <select size={marketsList.length} multiple={true} value={selectedAddedMarkets} onChange={(e) => this.handleSelectAddedMarket(Array.from(e.target.selectedOptions, option => option.value))}>
+                            {this.renderMarketList(addedMarkets)}
+                        </select>
+                    </div>
+                    <div className="setup-markets-block__list">
+                        <div className="setup-markets-block__list__command-add">
+                            <div className="setup-markets-block__list__command" onClick={this.handleAddMarket}>
+                                <ArrowLeftIcon /><span>Add</span>
```
Comment (josadcha) [inline src/components/SetupComponents/SetupMarketsBlock/index.tsx:62]: also missing translations in this file

File: src/modules/admin/markets/sagas/updateMarketSaga.ts [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        apiVersion: 'peatio',
+        headers: { 'X-CSRF-Token': csrfToken },
+    };
+};
+
+export function* updateMarketSaga(action: MarketUpdateFetch) {
+    try {
+        yield all(action.payload.map(item => call(API.post(enableMarketsConfig(getCsrfToken())), '/admin/markets/update', item)));
+
+        yield put(alertPush({message: ['Market updated created'], type: 'success'}));
```
Comment (josadcha) [inline src/modules/admin/markets/sagas/updateMarketSaga.ts:22]: add to translations please

**Review by josadcha — CHANGES_REQUESTED** [review]

**Review by josadcha — APPROVED** [review]

**Comment by oyershov** [comment]
Merged with note.
1. Refactor asap using functional approach
**2. Add translations**
3. Use separate color variables for the setup pages

**Pull request merged**